### PR TITLE
Enable CertificateCallbackThrowPropagates on Alpine

### PR DESCRIPTION
**Pull request by liveans**
We're using msquic main on Alpine images now, so let's test this test change over CI.

### CODE REVIEW

File: src/libraries/System.Net.Quic/tests/FunctionalTests/MsQuicTests.cs [hunk 1]
```diff
@@ -352,10 +352,14 @@ public async Task UntrustedClientCertificateFails()
             }
         }
 
-        [Fact]
-        [ActiveIssue("https://github.com/dotnet/runtime/issues/99074")]
+        [ConditionalFact]
```
Comment (ManickaP) [inline src/libraries/System.Net.Quic/tests/FunctionalTests/MsQuicTests.cs:355]:
```suggestion
        [ActiveIssue("https://github.com/dotnet/runtime/issues/99074", typeof(PlatformDetection), nameof(PlatformDetection.IsNotAlpine))]
```
You'd need to introduce `IsNotAlpine`, but that's trivial.

**Comment by rzikm** [comment]
I'd rather check for MsQuic version than for specific distro flavor.

**Review by ManickaP — APPROVED** [review]

**Pull request merged**